HousingSearchApi/V1/Helper/PostCodeHelpers.cs

```diff
@@ -28,10 +28,12 @@ public static bool SearchTextIsValidPostCode(string searchText)
 
             if (trimmed.Length > 7) return false;
 
-            const string pattern = @"^[A-Z]{1,2}[0-9]{1,2} ?[0-9][A-Z]{2}$";
+            const string pattern = @"^(([a-z][0-9]{1,2})|([a-z][a-hj-y][0-9]{1,2})|([a-z][0-9][a-z])|([a-z][a-hj-y][0-9]?[a-z]))[0-9][a-z]{2}$";
```

Comment:
Would it be worth adding a test coverage against known list of valid post code formats to ensure this captures them all?

Reply:
Yup, have done this now as a set fixture in the test file. Not sure if this is considered good practice or if I should use some sort of fixture generator. Let me know what you think.

Reply:
I think that setup is fine since it's hard to see where those values could be used elsewhere in tests. If we generate similar datasets to be used in multiple tests then it would make sense to create a generator for that.